src/controllers/admin/UserAjax.php

```diff
@@ -459,7 +460,234 @@ private function getRoleInfo($user)
+                if (!in_array($direction, $validDirection)) {
+                    throw new \Exception('The attribute used to direction is invalid.');
+                }
+                $data->direction = $direction;
+            }
+
+            $logModel = new Log();
+            $user = new User();
+            $basicAuth = new BasicAuthenticator($user);
+            $currentUser = $basicAuth->currentUser();
+
+            $responseLog = $logModel->getLogsByUserId($currentUser->getUserId(),$data->page, $data->page_size, $data->sort, $data->direction);
+
+            $output = array();
+            foreach ($responseLog->logs as $log) {
+                $output[] = array('id' => $log->getId(),
+                                  'event'      => $log->getEventLog(),
+                                  'event_data' => $log->getEventData(),
+                                  'created_at' => $log->getCreatedAt()
+                );
+            }
+            $response['success'] = true;
+            $response['user_id'] = $currentUser->getUserId();
+            $response['activities'] = $output;
+            $response['page'] = $data->page;
+            $response['page_size'] = $data->page_size;
+            $response['sort'] = $data->sort;
+            $response['direction'] = $data->direction;
+            $this->setStatusCode(200);
+        } catch (\Exception $e) {
+            $response['error_message'] = $e->getMessage();
+            $response['error_code'] = $e->getCode();
+        }
+
+        $this->setContent($response);
+    }
 
+    /**
+     *
+     * create a new event Log for current user
+     *
+     */
+    public function postAddUserEvent()
+    {
+        $response = array(
+            "method" => "adduserevent",
+            "success" => false,
+            "log" => "",
+            "user_id" => "",
+            "error_code" => 0,
+            "error_message" => ""
+        );
+
+        try {
+            $data = json_decode(file_get_contents("php://input"));
+            if (empty($data)) {
+                $data = (object) $_POST;
+            }
+            // Check required fields
+            $requiredParams = array('event');
+            $params = (array) $data;
+            foreach ($requiredParams as $param) {
+                if (empty($params[$param])) {
+                    throw new \Exception(ucfirst($param) .' is required.');
+                }
+            }
+
+            if(!array_key_exists("event_data", $params)){
+                $data->event_data = "";
+            }
+
+            $logModel = new Log();
+            $user = new User();
+            $basicAuth = new BasicAuthenticator($user);
+            $currentUser = $basicAuth->currentUser();
+
+            $frontEndClarify = '[front_end] '; //we need to clarify from where is the creation log
+            $logId = $logModel->create($currentUser->getUserId(), $frontEndClarify.$data->event, $data->event_data);
```

Comment:
this should be an array merge to add the data type the event, not an append. this will record bad JSON.